### PR TITLE
AlertingNG:  Test definition 

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Adds functionality to test a alert definition and get alert instances back.

**Special notes for your reviewer**:
To test: 
Add a query (A)
Add a reduce expression (B) on A to reduce the query to a single value
Add a math expression on B to compare the value to some other value.

Hit test.

### CODE REVIEW

File: public/app/features/alerting/state/reducers.ts [hunk 1]
```diff
@@ -62,11 +62,14 @@ export const initialAlertDefinitionState: AlertDefinitionState = {
...
   },
-  queryOptions: { maxDataPoints: 100, dataSource: { name: '-- Mixed --' }, queries: [] },
   queryRunner: new PanelQueryRunner(dataConfig),
   uiState: { ...store.getObject(ALERT_DEFINITION_UI_STATE_STORAGE_KEY, DEFAULT_ALERT_DEFINITION_UI_STATE) },
   data: [],
   alertDefinitions: [] as AlertDefinition[],
+  /* These are functions as they are mutated later on and redux toolkit will Object.freeze state so
+   * we need to store these using functions instead */
+  getInstances: () => [] as DataFrame[],
+  getQueryOptions: () => ({ maxDataPoints: 100, dataSource: { name: '-- Mixed --' }, queries: [] }),
```
Comment (torkelo) [inline public/app/features/alerting/state/reducers.ts:72]: What is mutable in the query options?

Comment (peterholmberg) [reply public/app/features/alerting/state/reducers.ts:72]: expr in prometheus queries

Comment (torkelo) [reply public/app/features/alerting/state/reducers.ts:72]: where is that mutated?

Comment (peterholmberg) [reply public/app/features/alerting/state/reducers.ts:72]: in the query editor.

Comment (torkelo) [reply public/app/features/alerting/state/reducers.ts:72]: Can't we change that? must be a mistake (or old way of doing things)

Comment (peterholmberg) [reply public/app/features/alerting/state/reducers.ts:72]: Sure, I can look into it.

Comment (torkelo) [reply public/app/features/alerting/state/reducers.ts:72]: But old angular editors do modify it, so an option would be to clone it before we give it to angular editors. But let's keep like this for now

**Review by torkelo — APPROVED** [review]

**Pull request merged**